src/canopy/knowledge_base/security_scanner/firewall.py

```diff
@@ -0,0 +1,66 @@
+import os
+
+import click
+import requests
+
+
+class AIFirewallError(ValueError):
+    pass
+
+
+class AIFirewall:
+
+    def __init__(self) -> None:
+        """Initialize the AI Firewall using required RI environment variables."""
+        self.firewall_api_key = self._get_env_var("FIREWALL_API_KEY")
+        self.firewall_url = self._get_env_var("FIREWALL_URL")
+        self.firewall_instance_id = self._get_env_var("FIREWALL_INSTANCE_ID")
+        self.firewall_instance_url = (
+            f"{self.firewall_url}/v1-beta/firewall/{self.firewall_instance_id}/validate"
+        )
+        self.firewall_headers = {
+            "X-Firewall-Api-Key": self.firewall_api_key.strip(),
+        }
+
+    @staticmethod
+    def _get_env_var(var_name: str) -> str:
+        env_var = os.environ.get(var_name)
+        if not env_var:
+            raise AIFirewallError(
+                f"{var_name} environment variable "
+                f"is required to use security scanning."
+            )
+        return env_var
+
+    def scan_text(self, text: str) -> None:
+        """Scan the input text for potential prompt injection attacks.
+
+        This method sends the input text to the AI Firewall via REST
+        API for security scanning.
+        """
```

Comment:
You might consider adding a link to relevant documentation about this API, so that users can understand more about how it works if they're interested.